shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -33,16 +33,19 @@ class JSBackend {
     * This function only translate name patterns and object patterns. I was thinking if we can
     * support literal parameter matching by merging multiple function `def`s into one.
     */
-  private def translatePattern(t: Term): JSPattern = t match {
+  private def translatePattern(t: Term)(implicit scope: Scope): JSPattern = t match {
     // fun x -> ... ==> function (x) { ... }
     // should returns ("x", ["x"])
-    case Var(name) => JSNamePattern(name)
+    case Var(name) => {
+      val runtimeName = scope.declareParameter(name)
+      JSNamePattern(runtimeName)
+    }
     // fun { x, y } -> ... ==> function ({ x, y }) { ... }
     // should returns ("{ x, y }", ["x", "y"])
     case Rcd(fields) =>
       JSObjectPattern(fields map {
-        case (Var(nme), (Var(als), _)) if nme === als => nme -> N
-        case (Var(nme), (subTrm, _))                  => nme -> S(translatePattern(subTrm))
+        case (Var(nme), (Var(als), _)) if nme === als => scope.declareParameter(nme);(); nme -> N
```

Comment:
Why not use the runtime name returned by `declareParameter` here?

Reply:
These are names of record fields, so I just declared and kept using the original names.

Reply:
Please don't resolve the conversation before I had a chance to see your answer. I would have missed this if I did not manually go through looking at all resolved comments.

> I just declared and kept using the original names

Are you certain the original name will be the same as the one assigned by `declareParameter`? Even if you are, future changes may break this assumption. So why not do the safe thing here?

Reply:
You didn't understand the problem.

I just noticed that @chengluyu independently reimplemented this same change (I don't know why), and he got the logic right. Please have a look:

https://github.com/hkust-taco/mlscript/pull/123/files/e20a99bf6815c0e6b12b9bf98957e13eb16e55b5..dc35036b3b2bef960e29c5eacec188c6c1979fc0#diff-47ab5806a2f49b005e96b51079228893fa2ee6c461349a441f99eb16bbcbd6dfL44-R56
```scala
        case (Var(nme), (Var(als), _)) if nme === als =>
          val runtimeName = scope.declareParameter(nme)
          if (JSField.isValidIdentifier(nme)) nme -> N
          else JSLit.makeStringLiteral(nme) -> S(JSNamePattern(runtimeName))
```